ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -652,75 +641,41 @@ blockBytes blk =
               CBOR.encodeWord 1
            <> CBOR.encodePreEncoded (CC.Block.blockAnnotation b)
 
--- | Encode a block using CBOR-in-CBOR tag 24.
-encodeByronBlock :: ByronBlockOrEBB cfg -> Encoding
-encodeByronBlock = encodeNestedCborBytes . blockBytes
-
--- | Inversion of `encodeByronBlock`. The annotation will be correct, because
--- the full bytes are available thanks to the CBOR-in-CBOR encoding.
-decodeByronBlock :: CC.Slot.EpochSlots -> Decoder s (ByronBlockOrEBB cfg)
-decodeByronBlock epochSlots = do
-  theBytes <- decodeNestedCborBytes
-  case decodeFullAnnotatedBytes "Block" internalDecoder (Lazy.fromStrict theBytes) of
-    Right it  -> pure $ ByronBlockOrEBB it
-    -- FIXME
-    --   err :: DecodeError
-    -- but AFAICT the only way to make the decoder fail is to give a `String`
-    -- to `fail`...
-    Left  err -> fail (show err)
+-- | Inversion of 'encodeByronBlock'. The annotation will be correct, because
+-- the full bytes are passed to the decoded value.
+decodeByronBlock :: CC.Slot.EpochSlots
+                 -> Decoder s (Lazy.ByteString -> ByronBlockOrEBB cfg)
+decodeByronBlock epochSlots =
+    fillInByteString <$> CC.Block.fromCBORABlockOrBoundary epochSlots
```

Comment:
Why not use `decodeFullAnnotatedBytes` rather than defining `fillInByteString`? Is it different?

Reply:
Oh yes I see, the types are different.

Reply:
At some point we should probably move it (possibly renamed) to where `decodeFullAnnotatedBytes` is defined. The `decodeFullAnnotatedBytes` is a bit over-specialised. There should be a function to just adjust the annotations from span to full bytes.